### PR TITLE
Search: Migrate some widget JS to a static file

### PR DESCRIPTION
**Pull request by Viper007Bond**
#### Changes proposed in this Pull Request:

* Move static JS to its own file so that it can be cached by the browser and possibly concatenated.
* Only load the JS/CSS if the widget is active.
* Rename admin JS to have -admin suffix.
* Remove inline HTML comment in favor of PHPDoc.

#### Testing instructions:

* Verify that the checkboxes on the front end still work.
* Verify that the JavaScript doesn't load when the widget isn't being used (`js/search-widget-filters.js` on the front end).
* Verify that the widget configuration in the customizer still works.

### CODE REVIEW

File: modules/search/class.jetpack-search-widget-filters.php [hunk 1]
```diff
@@ -248,19 +263,12 @@ public function widget( $args, $instance ) {
...
-		-->
 		<script type="text/javascript">
 				jQuery( document ).ready( function( $ ) {
-					var actionUrl      = <?php echo json_encode( home_url( '/' ) ); ?>,
-						orderByDefault = <?php echo json_encode( $orderby ); ?>,
+					var orderByDefault = <?php echo json_encode( $orderby ); ?>,
 						orderDefault   = <?php echo json_encode( $order ); ?>,
 						widgetId       = <?php echo json_encode( $this->id ); ?>,
-						currentSearch  = <?php echo json_encode( isset( $_GET['s'] ) ? $_GET['s'] : '' ); ?>
+						currentSearch  = <?php echo json_encode( isset( $_GET['s'] ) ? $_GET['s'] : '' ); ?>;
```
Comment (zinigor) [inline modules/search/class.jetpack-search-widget-filters.php]: I love where this is going. Can we go a step further maybe and move this bit of JS to the file too, so we can just use `wp_localize_script` to pass all these variables? We can check the `maybe` part inside the JS file.

File: modules/search/js/search-widget-filters.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	checkboxes.on( 'click change', function( e ) {
+		var anchor;
+		e.preventDefault();
+
+		anchor = jQuery( this ).closest( 'a' );
+		if ( anchor.length ) {
+			window.location.href = anchor.prop( 'href' );
+		}
+	} );
+} );
```
Comment (ebinnion) [inline modules/search/js/search-widget-filters.js]: Nitpick, we should have an empty newline at the end of the file.

File: modules/search/js/search-widget-filters.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+jQuery( document ).ready( function() {
```
Comment (gravityrail) [inline modules/search/js/search-widget-filters.js]: Typically with the document ready callback we accept the jQuery object as an arg called '$', like this:

```js
jQuery( document ).ready( function( $ ) { 
  //...
  anchor = $( this ).closest( 'a' );
  //...
}
```

Comment (ebinnion) [reply modules/search/js/search-widget-filters.js]: It's a bit different I think. `jQuery` is a global, and we pass it in to our script via an immediately invoked anonymous function (this is always a mouthful).

```js
( function( $ ) { 
    $( document ).ready( fn );
} ) ( jQuery ) ;
```

Comment (gravityrail) [reply modules/search/js/search-widget-filters.js]: I think both will work, based on a quick reading of the jQuery docs. Which we prefer is up to you. I see both styles in Jetpack - JITMs and Comment Like Counts use my style above, but IDC notices and the Jetpack Connection Banner use @ebinnion's style. I am going to approve this - it works anyway, and I don't see a reason to hold it up for this stylistic reason when there isn't a clear standard in the code.

File: modules/search/class.jetpack-search-widget-filters.php (hunk 1 above)
Comment (Viper007Bond) [reply modules/search/class.jetpack-search-widget-filters.php]: @zinigor I actually started doing exactly that but realized that the JS is unique per widget and weird things would likely happen if there were multiple copies of the widget being used at once. I decided to punt it for a future PR.

File: modules/search/js/search-widget-filters.js (hunk 2 above)
Comment (Viper007Bond) [reply modules/search/js/search-widget-filters.js]: Done.

**Pull request merged**